File: FoosballGames.WebApi/Controllers/ErrorsController.cs
using FoosballGames.Infrastructure;
using Microsoft.AspNetCore.Diagnostics;
using Microsoft.AspNetCore.Mvc;

namespace FoosballGames.WebApi.Controllers;

[ApiExplorerSettings(IgnoreApi = true)]
public class ErrorsController : ControllerBase
{
    [Route("error")]
    public IActionResult Error()
    {
        var context = HttpContext.Features.Get<IExceptionHandlerFeature>();
        var exception = context!.Error;
        var code = 500;

        if (exception is DomainException) code = 400;

        return Problem(context.Error.Message, statusCode: code);
    }
}

File: FoosballGames.WebApi/Controllers/ErrorsController.fs
namespace FoosballGames.WebApi.Controllers

open FoosballGames.Infrastructure
open Microsoft.AspNetCore.Diagnostics
open Microsoft.AspNetCore.Mvc

[<ApiExplorerSettings(IgnoreApi = true)>]
type ErrorsController() =
    inherit ControllerBase()
    
    [<Route("error")>]
    member this.Error() =
        let ctx = this.HttpContext.Features.Get<IExceptionHandlerFeature>()
        let ``exception`` = ctx.Error
        let code = if ``exception`` :? DomainException then 400 else 500
        this.Problem(ctx.Error.Message, statusCode = code)